### PR TITLE
Feature/rating movies and tv

### PR DESCRIPTION
**Pull request by marwanm7moud**

### CODE REVIEW

File: core/domain/usecase/src/main/java/com/chocolatecake/repository/MovieRepository.kt [hunk 1]
```diff
@@ -142,4 +144,9 @@ interface MovieRepository {
```
Comment (mohammed-akkad) [inline core/domain/usecase/src/main/java/com/chocolatecake/repository/MovieRepository.kt:145]: delete white space

File: feature/home/presentation/home-ui/src/main/res/layout/fragment_my_rated.xml [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+            android:layout_height="?android:attr/actionBarSize"
+            app:layout_constraintEnd_toEndOf="parent"
+            app:layout_constraintStart_toStartOf="parent"
+            app:layout_constraintTop_toTopOf="parent">
+
+            <ImageButton
+                android:id="@+id/backButton"
+                android:layout_width="24dp"
+                android:layout_height="24dp"
+                android:layout_marginStart="16dp"
```
Comment (mohammed-akkad) [inline feature/home/presentation/home-ui/src/main/res/layout/fragment_my_rated.xml]: use dimen :)

File: feature/home/presentation/home-ui/src/main/res/layout/fragment_my_rated.xml [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+                android:onClick="@{()->viewModel.onBackPressed()}"
+                android:scaleType="centerCrop"
+                android:src="@drawable/arrow_left"
+                android:layout_marginEnd="16dp"/>
+
+            <TextView
+                android:layout_width="wrap_content"
+                android:layout_height="wrap_content"
+                android:textColor="@color/on_background_87"
+                android:text="My Rated"
```
Comment (mohammed-akkad) [inline feature/home/presentation/home-ui/src/main/res/layout/fragment_my_rated.xml]: use extract string

**Review by mohammed-akkad — COMMENTED** [review]
nice job

File: core/data/repository/src/main/java/com/chocolatecake/repository/mappers/domain/my_rated/DomainMyRatedMoviesMapper.kt [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+import com.chocolatecake.entities.my_rated.MyRatedMovieEntity
+import com.chocolatecake.remote.response.dto.my_rated.MyRatedMovieDto
+import com.chocolatecake.repository.BuildConfig
+import com.chocolatecake.repository.mappers.Mapper
+import javax.inject.Inject
+
+class DomainMyRatedMoviesMapper @Inject constructor(private val genreEntities: List<GenreEntity>) :
+    Mapper<MyRatedMovieDto, MyRatedMovieEntity> {
+
+    override fun map(input: MyRatedMovieDto): MyRatedMovieEntity {
```
Comment (mosayed01) [inline core/data/repository/src/main/java/com/chocolatecake/repository/mappers/domain/my_rated/DomainMyRatedMoviesMapper.kt]:
```suggestion
class DomainMyRatedMoviesMapper @Inject constructor() :
    Mapper<MyRatedMovieDto, MyRatedMovieEntity> {

    override fun map(input: MyRatedMovieDto, genreEntities: List<GenreEntity>): MyRatedMovieEntity {
```

File: core/data/repository/src/main/java/com/chocolatecake/repository/mappers/domain/my_rated/DomainMyRatedTvShowMapper.kt [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+import com.chocolatecake.repository.mappers.Mapper
+import javax.inject.Inject
+
+class DomainMyRatedTvShowMapper @Inject constructor(
+    val genreEntities: List<GenreEntity>
+) :
+    Mapper<MyRatedTvShowDto, MyRatedTvShowEntity> {
+
+    override fun map(input: MyRatedTvShowDto): MyRatedTvShowEntity {
+        return MyRatedTvShowEntity(
```
Comment (mosayed01) [inline core/data/repository/src/main/java/com/chocolatecake/repository/mappers/domain/my_rated/DomainMyRatedTvShowMapper.kt]:
```suggestion
class DomainMyRatedTvShowMapper @Inject constructor() :
    Mapper<MyRatedTvShowDto, MyRatedTvShowEntity> {

    override fun map(input: MyRatedTvShowDto,  genreEntities: List<GenreEntity>): MyRatedTvShowEntity {
        return MyRatedTvShowEntity(
```

File: feature/home/presentation/viewmodel/src/main/java/com/chocolatecake/viewmodel/my_rated/MyRatedEvents.kt [hunk 6]
```diff
@@ -0,0 +1,14 @@
+package com.chocolatecake.viewmodel.my_rated
+
+
+sealed interface MyRatedEvents{
+    object OnBackPressed:MyRatedEvents
```
Comment (mosayed01) [inline feature/home/presentation/viewmodel/src/main/java/com/chocolatecake/viewmodel/my_rated/MyRatedEvents.kt]: Navigate back is better

**Review by mosayed01 — CHANGES_REQUESTED** [review]

**Review by mosayed01 — APPROVED** [review]

**Review by mohammed-akkad — APPROVED** [review]
great work

File: feature/home/presentation/home-ui/src/main/java/com/chocolatecake/ui/my_rated/MyRatedFragment.kt [hunk 7]
```diff
@@ -0,0 +1,58 @@
...
+    override fun onEvent(event: MyRatedEvents) {
+        when(event){
+            is MyRatedEvents.NavigateToMovieDetails -> findNavController().navigate(MyRatedFragmentDirections.actionMyRatedFragmentToMovieDetailsFragment(event.movieId))
+            is MyRatedEvents.NavigateToTVShowDetails -> findNavController().navigate(MyRatedFragmentDirections.actionMyRatedFragmentToMovieDetailsFragment(event.tvId))
+            is MyRatedEvents.NavigateBack -> findNavController().popBackStack()
+            is MyRatedEvents.ShowMyRatedMoviesPressed -> viewModel.getMyRatedMovies()
+            is MyRatedEvents.ShowMyRatedTvShowPressed -> viewModel.getMyRatedTvShow()
+        }
+    }
+
```
Comment (MohamedElgohary88) [inline feature/home/presentation/home-ui/src/main/java/com/chocolatecake/ui/my_rated/MyRatedFragment.kt]: white space :)

File: feature/home/presentation/viewmodel/src/main/java/com/chocolatecake/viewmodel/my_rated/MyRatedUiState.kt [hunk 8]
```diff
@@ -0,0 +1,20 @@
...
+import kotlinx.coroutines.flow.emptyFlow
+
+data class MyRatedUiState (
+    val MyRateType: RateType = RateType.Movies,
+    val MyRatedMedia: Flow<PagingData<MovieHorizontalUIState>> = emptyFlow(),
+    val errorList: List<String>? = emptyList(),
+    val isLoading: Boolean = false
+)
+
+
```
Comment (MohamedElgohary88) [inline feature/home/presentation/viewmodel/src/main/java/com/chocolatecake/viewmodel/my_rated/MyRatedUiState.kt]: white space :)

File: core/data/remote/src/main/java/com/chocolatecake/remote/response/dto/my_rated/MyRatedTvShowResponseDto.kt [hunk 9]
```diff
@@ -0,0 +1,15 @@
+package com.chocolatecake.remote.response.dto.my_rated
+
+
+import com.google.gson.annotations.SerializedName
+
+data class MyRatedTvShowResponseDto(
```
Comment (MohamedElgohary88) [inline core/data/remote/src/main/java/com/chocolatecake/remote/response/dto/my_rated/MyRatedTvShowResponseDto.kt]: what is the difference between this class and this class MyRatedTvShowDto

File: core/data/remote/src/main/java/com/chocolatecake/remote/response/dto/my_rated/MyRatedMovieDto.kt [hunk 10]
```diff
@@ -0,0 +1,37 @@
+package com.chocolatecake.remote.response.dto.my_rated
+
+
+import com.google.gson.annotations.SerializedName
+
+data class MyRatedMovieDto(
```
Comment (MohamedElgohary88) [inline core/data/remote/src/main/java/com/chocolatecake/remote/response/dto/my_rated/MyRatedMovieDto.kt:6]: you can use one class for the response 
two responses are the same

File: feature/home/presentation/viewmodel/src/main/java/com/chocolatecake/viewmodel/my_rated/MyRatedViewModel.kt [hunk 11]
```diff
@@ -0,0 +1,130 @@
...
+                        isLoading = false,
+                        errorList = emptyList()
+                    )
+                }
+            } catch (throwable: Throwable) {
+                onError(throwable)
+            }
+        }
+    }
+    fun getMyRatedTvShow(){
```
Comment (MohamedElgohary88) [inline feature/home/presentation/viewmodel/src/main/java/com/chocolatecake/viewmodel/my_rated/MyRatedViewModel.kt]: you can divide this function into small pieces 
and you can make a base function like as fetchRatings()

File: feature/home/presentation/viewmodel/src/main/java/com/chocolatecake/viewmodel/my_rated/MyRatedViewModel.kt [hunk 12]
```diff
@@ -0,0 +1,130 @@
...
+    }
+
+    private fun getData() {
+        when (_state.value.MyRateType) {
+            RateType.Movies -> getMyRatedMovies()
+            RateType.TVShows -> getMyRatedTvShow()
+        }
+    }
+
+    fun getMyRatedMovies(){
```
Comment (MohamedElgohary88) [inline feature/home/presentation/viewmodel/src/main/java/com/chocolatecake/viewmodel/my_rated/MyRatedViewModel.kt]: make this function private

Comment (marwanm7moud) [reply feature/home/presentation/viewmodel/src/main/java/com/chocolatecake/viewmodel/my_rated/MyRatedViewModel.kt]: i call it in fragment when i click on the chip

**Review by MohamedElgohary88 — APPROVED** [review]

**Pull request merged**